raiden/network/transport/matrix/transport.py

```diff
@@ -1337,10 +1337,10 @@ def _handle_to_device_message(self, event):
         return True
 
     def link_exception(self, callback: Any):
-        Runnable.link_exception(self, callback)
+        self.greenlet.link_exception(callback)
 
     def join(self, timeout=None):
-        Runnable.join(self, timeout)
+        self.greenlet.join(timeout)
```

Comment:
these were changed because the `link_exception` and `join` methods in the `Runnable` class work through the `__getattr__` hack that that class implements:
https://github.com/rsksmart/lumino/blob/af37b84c33e5e2a5c661551c19864cb96ec06ef8/raiden/utils/runnable.py#L75-L78

since these 2 methods are not formally defined on the `Runnable` class, the previous version of the code failed. this is, in principle, fixed by accessing the embedded greenlet directly.

Reply:
Not sure why using static calls was the first apporach but, if the class has a `greenlet ` member, it looks natural that it delegates the join call to that member.

Reply:
these weren't static calls; the code `ParentClass.func(self, ...)` is the equivalent of doing `super().func(...`, but directly specifying which parent class you want to use.

`super` will actually go through all the parent classes, which is both unnecessary and potentially problematic (remember this class actually has 2 parents outside of `Object`).